compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -44,14 +44,12 @@ class StringInterpolatorOpt extends MiniPhase {
   /** Matches an s or raw string interpolator */
   private object SOrRawInterpolator {
     def unapply(tree: Tree)(implicit ctx: Context): Option[(List[Literal], List[Tree])] = {
-      if (tree.symbol.eq(defn.StringContextRaw) || tree.symbol.eq(defn.StringContextS)) {
-        tree match {
-          case Apply(Select(Apply(StringContextApply(), List(Literals(strs))), _),
-          List(SeqLiteral(elems, _))) if elems.length == strs.length - 1 =>
-            Some(strs, elems)
-          case _ => None
-        }
-      } else None
+      tree match {
```

Comment:
A minor issue: the object name `SOrRawInterpolator` doesn't strictly correspond to what is in the code.

Reply:
That is not logic I made, I did not choose that name. Do you want me to change it to something else?

Reply:
It used to reflect what was in the code. Not anymore. I would just inline this extractor in `StringContextIntrinsic`

Reply:
Before the change, it's. After the symbolic equality checks are removed, it wasn't. But as we are going to remove local optimisations, it's fine to keep it as it is now.

Reply:
@liufengyun This is not part of the local optimisations that will be removed. Scalac does something similar